src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -32,10 +37,72 @@ def _ensure_ffi_initialized(cls):
         with cls._init_lock:
             if not cls._lib_loaded:
                 cls._lib_loaded = True
-                res = cls.lib.Cryptography_add_osrandom_engine()
+                res = cls._register_osrandom_engine()
                 assert res != 0
 
     @classmethod
+    def _register_osrandom_engine(cls):
+        if cls._retained:
+            return 2
+
+        def retain(it):
+            cls._retained.append(it)
+            return it
+        method = cls.ffi.new("RAND_METHOD*")
+        retain(method)
+        method.seed = cls.ffi.NULL
+
+        @retain
+        @cls.ffi.callback("int (*)(unsigned char *, int)", error=0)
+        def osrandom_rand_bytes(buf, size):
```

Comment:
I believe all these callbacks can just be class methods with the decorator, which will remove the need for a keepalive list.